src/wazuh_modules/wm_sca.c

```diff
@@ -3193,3 +3204,63 @@ static int append_msg_to_vm_scat (wm_sca_t * const data, const char * const msg)
     }
     return 0;
 }
+
+/*
+    Thread that sends the integrity hash to the manager every configurable number of seconds.
+    If the manager has a different integrity hash, it will request the agent to send a new
+    sca scan and integrity hash.
+ */
+static void *wm_sca_check_integrity_periodically (wm_sca_t * data) {
+    unsigned int time_sleep = data->integrity_interval;
+
+    while (1) {
+        /* Thread sleeps for 5 minutes until the next time it sends the hash */
+        mdebug2("Sleeping for %d seconds.", (int)time_sleep);
+        wm_delay(1000 * time_sleep);
+
+        /* It doesn't send anything if there are no policies */
+        if (!data->policies) {
+            continue;
+        }
+
+        int i;
+
+        /* Send hash for every policy file */
+        for (i = 0; data->policies[i]; i++) {
+            if (!data->policies[i]->enabled || !data->policies[i]->policy_id) {
+                continue;
+            }
+
+            cJSON *json_integrity = cJSON_CreateObject();
+            int cis_db_index = i;
+
+            cJSON_AddStringToObject(json_integrity, "type", "integrity_check");
+
+            mdebug1("Calculating hash for scanned results.");
+            
+            /* Mutex for global varible cis_db_for_hash */
+            w_mutex_lock(&control_msg_mutex);
+            char *integrity_hash = wm_sca_hash_integrity(cis_db_index);
+            w_mutex_unlock(&control_msg_mutex);
+
+            /* If there is no hash in the local db, it will send an empty one */
```

Comment:
If the hash is NULL it is because the policy is being skipped in the scans and shall not be sent to the manager.